### PR TITLE
Add/cors support

### PR DESCRIPTION
**Pull request by digitalsadhu**
Adds cors but by default its set to *

Configuration can be done in environment.js
```js
module.exports = function (environment) {
  const ENV = {
    host: 'http://localhost',
    port: 3010
  }

  ENV.cors = {
    origin: ENV.host,
    preflight: true
  }

  ENV.bodyparser = {
    json: {type: 'application/json', extended: false},
    text: {type: 'text/plain', extended: false},
    urlencoded: {type: 'application/x-www-form-urlencoded', extended: false},
    raw: {type: 'application/octet-stream', extended: false}
  }

  return ENV
}

```
Config options taken directly from the https://www.npmjs.com/package/cors project

### CODE REVIEW

**Comment by digitalsadhu** [comment]
@pnw thoughts?

File: src/classes/application.js [hunk 1]
```diff
@@ -160,6 +161,11 @@ module.exports = class Application extends Base {
     const app = express()
     _app.set(this, app)
 
+    log.trace('Ash server loading cors middleware')
+    if (config.cors !== false) {
```
Comment (pnw) [inline src/classes/application.js]: See my comment in https://github.com/ash-framework/ash/pull/4 - I think you need to check this value better.

10/10 if my config is:

```js
module.exports = function (environment) {
  const ENV = {
    port: 3010
  }

  return ENV
}
```

then this block becomes

```js
if (config.cors !== false) { // undefined !== false -> true
  app.use(cors(Object.assign({}, undefined)))
}
```

File: yarn.lock [hunk 2]
```diff
@@ -2670,3 +2707,7 @@ yeast@0.1.2:
   version "0.1.2"
   resolved "https://registry.yarnpkg.com/yeast/-/yeast-0.1.2.tgz#008e06d8094320c372dbc2f8ed76a0ca6c8ac419"
 
+yuidoc-ember-theme@^1.2.1:
+  version "1.2.1"
+  resolved "https://registry.yarnpkg.com/yuidoc-ember-theme/-/yuidoc-ember-theme-1.2.1.tgz#075274507797cf4166a8bedeb48f964a077a300b"
+
```
Comment (pnw) [inline yarn.lock:2713]: Holy shit turns out yarn config files are yuuuge..

0/10 shitpost

**Review by pnw — CHANGES_REQUESTED** [review]
Requesting better config option handling

File: src/classes/application.js [hunk 3]
```diff
@@ -160,6 +161,14 @@ module.exports = class Application extends Base {
     const app = express()
     _app.set(this, app)
 
+    log.trace('Ash server loading cors middleware')
+    if (typeof config.cors === 'object') {
+      if (config.cors.preFlight === true) {
```
Comment (pnw) [inline src/classes/application.js:166]: What's preflight?

Comment (digitalsadhu) [reply src/classes/application.js:166]: The options handling for cors where you use options to ask what cors stuff is supported.

**Pull request merged**